### PR TITLE
Updated the devfile metadata to be more consistent

### PR DESCRIPTION
**Pull request by schultzp2020**
### What does this PR do?:

1. Updates the devfile metadata to be more consistent i.e. capitalize Java in the language property.
2. Adds editorconfig and gitattributes
3. Adds github action concurrency check

### Which issue(s) this PR fixes:

N/A

### PR acceptance criteria:

- [x] Contributing guide
_Have you read the [devfile registry contributing guide](https://github.com/devfile/registry/blob/main/CONTRIBUTING.md) and followed its instructions?_
- [x] Test automation
_Does this repository's tests pass with your changes?_
- [ ] Documentation
_Does any documentation need to be updated with your changes?_
- [x] Check Tools Provider
_Have you tested the changes with existing tools, i.e. Odo, Che, Console? (See [devfile registry contributing guide](https://github.com/devfile/registry/blob/main/CONTRIBUTING.md) on how to test changes)_

### How to test changes / Special notes to the reviewer:

### CODE REVIEW

**Comment by schultzp2020** [comment]
Please review your stacks. Thanks!
@kadel 
@ajm01 
@maxandersen 
@ehsavoie 
@BethGriggs

**Review by ehsavoie — APPROVED** [review]
For java-wildfly-bootable and java-wildfly

**Comment by schultzp2020** [comment]
@maxandersen and @BethGriggs please review my changes before its planned merge date of 2022/10/13.

File: extraDevfileEntries.yaml
Comment (maxandersen) [inline extraDevfileEntries.yaml:23]: is the type not still lower case `quarkus`?

File: stacks/java-openliberty/devfile.yaml
Comment (scottkurz) [inline stacks/java-openliberty/devfile.yaml:21]: What would be the downside of leaving the version as-is for the four Liberty (Open/WebSphere , Maven/Gradle) devfiles?

We started down a path of using our own versioning strategy integrating our outer loop artifacts.   Following that to its conclusion we'd want to make other changes.

But if this change is just syntax cleanup, well, I think we could say this is equivalent to our previous version, e.g. 0.8.1 in this case, and merge in the change.   

(And we could open an issue in our devfile source repo to make the same changes there:  https://github.com/OpenLiberty/devfile-stack/issues ).   WDYT?

**Review by scottkurz — CHANGES_REQUESTED** [review]
Do we need to bump version?

File: extraDevfileEntries.yaml
Comment (schultzp2020) [reply extraDevfileEntries.yaml:23]: `Language` and `projectType` should follow the same case as the commercial display name.

File: stacks/java-openliberty/devfile.yaml
Comment (elsony) [reply stacks/java-openliberty/devfile.yaml:21]: In general, we recommend version updates (semver) when the stack is updated so that users know the underlying stack has been changed. Having said that, it is just a recommendation so it is up to the stack owner to have the final decision on the version number.

Comment (scottkurz) [reply stacks/java-openliberty/devfile.yaml:21]: I'd rather just leave the version alone in this PR then.

I opened: https://github.com/OpenLiberty/devfile-stack/issues/212 to follow-up and make these changes.  We can still view that repo as the source of record of our devfile.

But I think it's OK to go ahead and merge the rest of the changes.  We might not get to it for a few weeks so don't want to keep you waiting.   

There's an opportunity for minor user confusion here, I know, but unless there's some specific reason (like some version of some devfile tooling having a bug in the older syntax) then I don't think it will end up mattering all that much.

**Review by johnmcollier — APPROVED** [review]

**Comment by kadel** [comment]
The tests with odo v3 were failing, but it still got merged :-(

Now the dotnet devfiles are not working with odo.
This is a bug in odo (https://github.com/redhat-developer/odo/issues/6234), but still, this is precisely the reason we have tests here, so we know when something gets broken, and we can fix it.

If we ignore failing tests then I don't know why we even have them.

**Comment by johnmcollier** [comment]
@kadel Hey, sorry about that, you’re absolutely right, and merging this is in was premature. We had discussed this PR on a team call yesterday,, and were under the impression it would be safe to merge in, but that was clearly a mistake.

It also appears that the odo v3 tests are not a required check to merge, and that should definitely change.

**Pull request merged**
